core/templates/dev/head/domain/exploration/FallbackObjectFactory.js

```diff
@@ -18,37 +18,39 @@
  */
 
 oppia.factory('FallbackObjectFactory', [
-  'OutcomeObjectFactory',
-  function(OutcomeObjectFactory) {
+  'OutcomeObjectFactory', 'TriggerObjectFactory',
+  function(OutcomeObjectFactory, TriggerObjectFactory) {
     var Fallback = function(trigger, outcome) {
       this.trigger = trigger;
       this.outcome = outcome;
     };
 
     Fallback.prototype.toBackendDict = function() {
       return {
-        trigger: this.trigger,
+        trigger: this.trigger.toBackendDict(),
         outcome: this.outcome.toBackendDict()
       };
     };
 
     Fallback.createFromBackendDict = function(fallbackBackendDict) {
       return new Fallback(
-        fallbackBackendDict.trigger,
+        TriggerObjectFactory.createFromBackendDict(
+          fallbackBackendDict.trigger),
         OutcomeObjectFactory.createFromBackendDict(
           fallbackBackendDict.outcome));
     };
 
     Fallback.createDefault = function(dest) {
       return new Fallback(
-        {
-          trigger_type: 'NthResubmission',
-          customization_args: {
-            num_submits: {
-              value: 3
+        TriggerObjectFactory.createFromBackendDict(
+          {
+            trigger_type: 'NthResubmission',
+            customization_args: {
+              num_submits: {
+                value: 3
+              }
             }
-          }
-        },
+          }),
```

Comment:
(Optional) It's a bit weird that the indentation is different between this and the next line. Perhaps you could do:

```
  return new Fallback(
    TriggerObjectFactory.createFromBackendDict({
      trigger_type: ...,
      ...
    }),
    OutcomeObjectFactory.createNew(...)
  );
```